packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -134,16 +137,27 @@ export const NotificationsContextProvider = ({
       return 'denied';
     }
 
+    notificationSourceRef.current = source;
     const result = await globalThis.window?.Notification?.requestPermission();
-    trackEvent({
-      event_name: AnalyticsEvent.ClickEnableNotification,
-      extra: JSON.stringify({ origin: source, permission: result }),
-    });
     await onUpdatePermission(result);
 
     return result;
   };
 
+  useEffect(() => {
+    subscriptionCallbackRef.current = (isSubscribedNew) => {
+      if (isSubscribedNew) {
+        trackEvent({
+          event_name: AnalyticsEvent.ClickEnableNotification,
+          extra: JSON.stringify({
+            origin: notificationSourceRef.current,
+            permission: 'granted',
```

Comment:
Its not always granted right? Think it also triggers on denied

Reply:
Yap, this is why it's inside a condition:
`if (isSubscribedNew) {`